oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/ElasticResultRowAsyncIterator.java

```diff
@@ -111,6 +111,21 @@ public boolean hasNext() {
                 throw new IllegalStateException("Error reading next result from Elastic", e);
             }
         }
+
+        // Check if there are any Throwable filled from onFailure Callback in the errorReference
+        // Any exception (such as ParseException) during the prefetch (init scanner) via the async call to ES would be available here
+        // when the cursor is actually being traversed.
+        // This is being done so that we can log the caller stack trace in case of any exception from ES and not just the trace of the async query thread.
+
+        Throwable error = errorRef.get();
```

Comment:
The `errorRef` is not cleared by this call to `get`. Is there the risk that this method will be called again and it will log the same exception several times? Maybe use `getAndSet(null)` to ensure we do not log more than once.

Reply:
Thanks. Good catch.